src/Build/NuGet.Build.Packaging.Tasks/NuGet.Build.Packaging.targets

```diff
@@ -369,14 +361,43 @@ Copyright (c) .NET Foundation. All rights reserved.
 		</AssignPackagePath>
 	</Target>
 
+	<!-- Separate out the final package file, which is a description of the package itself, into a separate target.
+		 This allows users to craft the description of the package (version, id, etc.) based on the contents if they so choose. 
+	-->
+	<PropertyGroup>
+		<GetFinalPackageContentsDependsOn>
+			$(GetFinalPackageContentsDependsOn);
+			GetPackageContents;
+			GetPackageTargetPath;
+		</GetFinalPackageContentsDependsOn>
+	</PropertyGroup>
+	<Target Name="GetFinalPackageContents" DependsOnTargets="$(GetFinalPackageContentsDependsOn)" Returns="@(_PackageContent)">
+		<!-- If packaging the project, provide the metadata as a non-file item -->
+		<ItemGroup Condition="'$(PackageId)' != ''">
+			<_SelfReferentialPackageFile Include="@(PackageTargetPath->'%(Id)')">
+				<Kind>Metadata</Kind>
+				<PackageId>$(PackageId)</PackageId>
+				<Platform>$(Platform)</Platform>
+				<TargetFrameworkMoniker Condition="'$(IsPackagingProject)' != 'true'">$(TargetFrameworkMoniker)</TargetFrameworkMoniker>
+			</_SelfReferentialPackageFile>
+		</ItemGroup>
+
+		<AssignPackagePath Files="@(_SelfReferentialPackageFile)"
```

Comment:
strictly speaking, I don't think the metadata item needs to be assigned a package path at all. Meaning this target could be just:

```
<ItemGroup>
  <_PackageContent Include="@(PackageTargetPath->'%(Id)'">
  	<Kind>Metadata</Kind>
    <PackageId>$(PackageId)</PackageId>
    <Platform>$(Platform)</Platform>
    <TargetFrameworkMoniker Condition="'$(IsPackagingProject)' != 'true'">$(TargetFrameworkMoniker)</TargetFrameworkMoniker>
  </_PackageContent>
</ItemGroup>
```

The reason is that the `GetPackageTargetPath` is already checking for the condition on this item group.